sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -583,6 +583,66 @@ class QueryCompilationErrorsSuite
       msg = "The deserializer is not supported: try to map \"STRUCT<a: STRING, b: INT>\" " +
         "to Tuple1, but failed as the number of fields does not line up.")
   }
+
+  test("UNSUPPORTED_GENERATOR: " +
+    "generators are not supported when it's nested in expressions") {
+    val e = intercept[AnalysisException](
+      sql("""select explode(Array(1, 2, 3)) + 1""").collect()
+    )
+
+    checkErrorClass(
+      exception = e,
+      errorClass = "UNSUPPORTED_GENERATOR",
+      errorSubClass = Some("NESTED_IN_EXPRESSIONS"),
+      msg = "The generator is not supported: nested in expressions (explode(array(1, 2, 3)) + 1)")
```

Comment:
Please, quote (explode(array(1, 2, 3)) + 1) by default (double quotes). Maybe, it is better to add a helper method `toSQLExpr()` to QueryErrorsBase